### PR TITLE
Redirect non-mappers to Reservations Instructions page

### LINKED ISSUE #1198: 403 forbidden if try to access reservations page without individual mapper privileges
Users should be able to view the page but not make reservations.

### PR DESCRIPTION
**Pull request by kdeloach**
When non individual mappers try to access the Reservations page, they
will be redirected to the reservations instructions page.

Fixes #1198

### CODE REVIEW

File: src/tiler/sql/user_reservable.sql [hunk 1]
```diff
@@ -46,4 +62,7 @@
     ON (turf.group_id = trustedmapper.group_id
         AND trustedmapper.user_id = <%= user_id %>
         AND trustedmapper.is_approved)
+  LEFT JOIN core_user AS individual_mapper
+    ON (individual_mapper.id = <%= user_id %>
+        AND COALESCE(individual_mapper.individual_mapper, FALSE))
```
Comment (maurizi) [inline src/tiler/sql/user_reservable.sql]: Does this Model need to be added to the cache buster?

Comment (kdeloach) [inline src/tiler/sql/user_reservable.sql]: In my testing, the available blockfaces on the reservables page updated immediately as I toggled the Individual Mapper flag on & off. I'll add the User table to the reservable cache buster for good measure.

**Comment by kdeloach** [comment]
Added user model to cache buster.

**Comment by kdeloach** [comment]
@jwalgran This could use some more eyes.

**Comment by jwalgran** [comment]
The most I can give you is 2

**Comment by dboyer** [comment]
Doesn't a user have to have individual mapper status before they can request to map in a group's area? I don't see the request individual mapper status button on a group's detail page unless I'm already an individual mapper.

**Comment by kdeloach** [comment]
Yes, you must be an individual mapper before requesting group mapper status. The only way to become an individual mapper is by clicking a button on your dashboard/profile once you have fulfilled certain requirements.

Maybe I overcomplicated this though. It seems like instead of showing the Reservations map for non-mappers, we should just immediately redirect to the Reservations Instructions page? We could also add a "Become a trusted mapper" button on the group detail page which links to the instructions page. What do you think?

**Comment by maurizi** [comment]
> It seems like instead of showing the Reservations map for non-mappers, we should just immediately redirect to the Reservations Instructions page? We could also add a "Become a trusted mapper" button on the group detail page which links to the instructions page. What do you think?

+1 to this

**Comment by dboyer** [comment]
@kdeloach Parks approved redirecting to the reservations instructions page when a non-individual mapper clicks the reservations link in the navigation. Non-individual mappers don't need to see the reservations map. 

The group detail includes a request individual mapper status within the group area button if that group allows individual mappers.

**Comment by kdeloach** [comment]
Thanks, that makes sense to me. I'll make these changes tomorrow.

**Comment by kdeloach** [comment]
Updated. Will reassign once tests have passed.

**Comment by maurizi** [comment]
@kdeloach You may want to update the PR title now.

**Comment by maurizi** [comment]
I'll look at this shortly

**Comment by maurizi** [comment]
@kdeloach It looks like there is some lint in this PR.

**Comment by maurizi** [comment]
I tried to go to the reservations map as a user w/o individual mapper status and got redirected to the individual mapper instructions page.
When I set the individual mapper flag to True I was able to get to the reservations page.

The process for requesting trusted mapper status mostly worked great.
The other thing weird was that after I requested trusted mapper status, when I went back to the group detail page there was no change indicating that a request had been sent, and I could continue to request trusted mapper status.

However, I think that is a small issue that doesn't need to be fixed right now.

+1

**Comment by kdeloach** [comment]
Fixed lint.

**Comment by kdeloach** [comment]
Once you are approved as a trusted mapper, that button will actually forward to the Reservations page instead of submitting another request (not that that is any less confusing).

**Pull request merged**